File: python/015.py
#!/usr/bin/env python3

def main():
  """ this computes projecteuler.net problem 015 """

# need to compute 2n!/((n!)*(n!))

  limit = 20
  numerator = []
  denominator = []
  for i in range(limit):
    denominator.append(i+1)
    numerator.append(i+limit+1)

#  walk through lists and cast away multiples to reduce size of numbers from
#  factorials - this is incomplete factoring but sufficient for this problem
#  we could factor both completely if we wanted...

  for i in (denominator):
    for j in (numerator):
      if ((i != 1) and ((j % i) == 0)):
        j = j / i
        i = 1

  num = 1
  denom = 1
  for i in range(limit):
    num *= numerator[i]
    denom *= denominator[i]
      
  print(num//denom)
  return 0

if __name__ == "__main__":
    main()

File: perl6/015.pl
#!/usr/bin/perl6

# need to compute 2n!/((n!)*(n!))

my $limit = 20;
my (@numerator, @denominator);
loop (my $i=0; $i < $limit; $i++) {
  push @denominator, $i+1;
  push @numerator, $i+$limit+1;
}

#  walk through lists and cast away multiples to reduce size of numbers from
#  factorials - this is incomplete factoring but sufficient for this problem
#  we could factor both completely if we wanted...

for (@denominator) -> $i {
  my $n = $i;
  for (@numerator) -> $j {
    my $k = $j;
    if (($n != 1) and (($k % $n) == 0)) {
      $k = $k / $n;
      $n = 1
    }
  }
}

my $num = 1;
my $denom = 1;
loop (my $m = 0; $m < $limit; $m++) {
  $num *= @numerator[$m];
  $denom *= @denominator[$m];
}
    
say $num/$denom;
